### PR TITLE
Configure workflows file path in the token

### PR DESCRIPTION
**Pull request by eduardoj**
Add a field in the token form to configure a custom path for the workflows configuration file.

Peek 2022-12-09 13-00_configure_workflows_file_path_in_the_token

### For reviewers

Make sure you have enabled the `trigger_workflow` feature flag.

### CODE REVIEW

File: src/api/app/views/webui/users/tokens/edit.html.haml [hunk 1]
```diff
@@ -47,6 +47,13 @@
                     = f.password_field(:scm_token, size: 80, class: 'form-control', placeholder: 'Please enter your new SCM token')
                   .form-text.text-muted
                     Leaving this text field empty will not update your SCM token.
+                .form-group#scm-path-group
+                  = f.label(:scm_path) do
+                    Custom Path for Workflows Configuration File:
+                  .input-group
+                    = f.text_field(:scm_path, class: 'form-control', placeholder: 'Eg: .my_subdir/my_workflows_file.yml')
+                  .form-text.text-muted
+                    Leaving this text field empty will use the default value '.obs/workflows.yml'
```
Comment (krauselukas) [inline src/api/app/views/webui/users/tokens/edit.html.haml:56]:
```suggestion
                    Leaving this text field empty will use the default path '.obs/workflows.yml'
```

File: src/api/app/views/webui/users/tokens/new.html.haml [hunk 2]
```diff
@@ -38,12 +38,19 @@
...
                 = f.label(:scm_token, 'SCM Token:')
                 %abbr.text-danger{ title: 'required' } *
                 = f.password_field(:scm_token, size: 80, class: 'form-control', placeholder: 'Please enter your SCM token')
+              .form-group
+                = f.label(:scm_path) do
+                  Custom Path for Workflows Configuration File:
+                .input-group
+                  = f.text_field(:scm_path, class: 'form-control', placeholder: 'Eg: .my_subdir/my_workflows_file.yml')
+                .form-text.text-muted
+                  Leaving this text field empty will use the default value '.obs/workflows.yml'
```
Comment (krauselukas) [inline src/api/app/views/webui/users/tokens/new.html.haml:53]:
```suggestion
                  Leaving this text field empty will use the default path '.obs/workflows.yml'
```

File: src/api/app/views/webui/users/tokens/edit.html.haml (hunk 1 above)
Comment (eduardoj) [reply src/api/app/views/webui/users/tokens/edit.html.haml]: Applied.

File: src/api/app/views/webui/users/tokens/new.html.haml (hunk 2 above)
Comment (eduardoj) [reply src/api/app/views/webui/users/tokens/new.html.haml]: Applied.

File: src/api/db/migrate/20221208125517_add_scm_path_to_tokens.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class AddScmPathToTokens < ActiveRecord::Migration[7.0]
+  def change
+    add_column :tokens, :scm_path, :string, default: ''
```
[the remark this answers is not in the document]
Comment (eduardoj) [reply src/api/db/migrate/20221208125517_add_scm_path_to_tokens.rb]: I opted for `workflow_configuration_path`. Please have a look again. I hope I haven't forgotten any change.

**Comment by krauselukas** [comment]
@eduardoj Changes looking good, just one thing, maybe we should show the path on the token index as well?

**Comment by krauselukas** [comment]
> > @eduardoj Changes looking good, just one thing, maybe we should show the path on the token index as well?
> 
> I also thought about that, but most of the times it will have a repeated value. And when not the value of the path could overload the width easily. If you thing it provides enough value as to show it there, I could modify the code. But we could add it later on.

Fine for me

**Review by krauselukas — APPROVED** [review]

**Pull request merged**